web/src/features/auth/components/MoreCast2AuthWrapper.tsx

```diff
@@ -0,0 +1,60 @@
+import * as jwtDecode from 'jwt-decode'
+import React, { useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { AppDispatch } from 'app/store'
+import { TEST_AUTH, WF1_AUTH_URL } from 'utils/env'
+import { wf1Authenticate, wf1AuthenticateError } from 'features/auth/slices/wf1AuthenticationSlice'
+import { selectAuthentication, selectWf1Authentication } from 'app/rootReducer'
+import { ROLES } from 'features/auth/roles'
+
+interface Props {
+  children: React.ReactElement
+}
+
+const MoreCast2AuthWrapper = ({ children }: Props) => {
+  const dispatch: AppDispatch = useDispatch()
+
+  const [renderChildren, setRenderChildren] = useState(false)
+  const { isAuthenticated, roles } = useSelector(selectAuthentication)
```

Comment:
I don't think we need `isAuthenticated` as this wrapper component won't even be rendered if we're not authenticated. We could look at it and remove in another PR.

Reply:
Good call, removed. in https://github.com/bcgov/wps/pull/2854/commits/130a77fb077cc13f4c90926b4ddce7c7f2f8dd99